core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -833,10 +850,12 @@ private[history] class AppListingListener(log: FileStatus, clock: Clock) extends
     var sparkUser: String = null
     var completed = false
     var appSparkVersion = ""
+    val jobToStatus = new mutable.HashMap[Int, String]
```

Comment:
I suggest that the value type can be Boolean instead of String, or use a HashSet to mark either the successful ones or failed ones.